libpng dependency it's an issue for Mac users

The issue with libpng was opened last year and the problems persist (https://github.com/pkrumins/node-png/issues/42).
This problem seems to be well known (https://www.npmjs.com/package/depthcam).
Building it from source didn't solve nothing.

I'd suggest using another png encoder lib than libpng.

Nice job, btw.

The following are the errors in node-png module installation:

``````
../src/common.cpp:10:17: error: calling a protected constructor of class
      'v8::HandleScope'
    HandleScope scope;
                ^
/Users/fritz/.node-gyp/0.11.13/deps/v8/include/v8.h:845:13: note:
      declared protected here
  V8_INLINE HandleScope() {}
            ^
../src/common.cpp:11:37: error: no member named 'New' in 'v8::String'
    return Exception::Error(String::New(msg));
                            ~~~~~~~~^
../src/common.cpp:16:17: error: calling a protected constructor of class
      'v8::HandleScope'
    HandleScope scope;
                ^
/Users/fritz/.node-gyp/0.11.13/deps/v8/include/v8.h:845:13: note:
      declared protected here
  V8_INLINE HandleScope() {}
            ^
../src/common.cpp:17:12: error: use of undeclared identifier 'ThrowException';
      did you mean 'ErrorException'?
    return ThrowException(ErrorException(msg));
           ^~~~~~~~~~~~~~
           ErrorException
../src/common.cpp:8:1: note: 'ErrorException' declared here
ErrorException(const char *msg)
^
../src/common.cpp:17:27: error: no viable conversion from 'Handle<v8::Value>' to
      'const char *'
    return ThrowException(ErrorException(msg));
                          ^~~~~~~~~~~~~~~~~~~
../src/common.cpp:8:28: note: passing argument to parameter 'msg' here
ErrorException(const char *msg)
                           ^
5 errors generated.
make: *** [Release/obj.target/png/src/common.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1046:12)
gyp ERR! System Darwin 14.0.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/fritz/Desktop/shader-test-html/node_modules/web-bundle/node_modules/png
gyp ERR! node -v v0.11.13
gyp ERR! node-gyp -v v0.13.0
gyp ERR! not ok
npm ERR! png@3.0.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the png@3.0.3 install script.
npm ERR! This is most likely a problem with the png package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls png
npm ERR! There is likely additional logging output above.

npm ERR! System Darwin 14.0.0
npm ERR! command "/usr/local/bin/node" "/usr/local/bin/npm" "install" "--save" "web-bundle"
npm ERR! cwd /Users/fritz/Desktop/shader-test-html
npm ERR! node -v v0.11.13
npm ERR! npm -v 1.4.9
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /Users/fritz/Desktop/shader-test-html/npm-debug.log
npm ERR! not ok code 0```
``````

--- a/package.json
+++ b/package.json
@@ -22,11 +22,9 @@
   "dependencies": {
     "async": "^0.9.0",
     "node-png": "^0.4.3",
-    "png-js": "^0.1.1",
     "yargs": "^3.7.0"
   },
-  "contributors":
-	[
-	"Joe Dytrych <j@dytry.ch> http://j.dytry.ch/"
-	]
+  "contributors": [
+    "Joe Dytrych <j@dytry.ch> http://j.dytry.ch/"
+  ]
 }

--- a/package.json
+++ b/package.json
@@ -21,7 +21,7 @@
   },
   "dependencies": {
     "async": "^0.9.0",
-    "png": "^3.0.3",
+    "node-png": "^0.4.3",
     "png-js": "^0.1.1",
     "yargs": "^3.7.0"
   },

--- a/tool/Bundle.js
+++ b/tool/Bundle.js
@@ -1,6 +1,5 @@
 var fs = require('fs'),
   path = require('path'),
-  PNG = require('png-js'),
   nodePng = require('node-png');
 
 function Bundle(key) {
@@ -25,13 +24,13 @@ Bundle.prototype.files = function() {
 // Load a png file and decode it
 Bundle.prototype.load = function(filename, cb) {
   var self = this;
-  fs.readFile(filename, function(err, data) {
-    if (err) return cb(err);
-    (new PNG(data)).decode(function(pixels) {
-      self.decode(stripAlpha(pixels));
+  fs.createReadStream(filename)
+    .pipe(new nodePng.PNG())
+    .on('parsed', function() {
+      self.decode(stripAlpha(this.data));
       cb();
-    });
-  });
+    })
+    .on('error', cb);
 };
 
 // Decode a pixel buffer

--- a/tool/Bundle.js
+++ b/tool/Bundle.js
@@ -1,7 +1,7 @@
 var fs = require('fs'),
   path = require('path'),
   PNG = require('png-js'),
-  PngEncoder = require('png').Png;
+  nodePng = require('node-png');
 
 function Bundle(key) {
   if (key) {
@@ -97,27 +97,29 @@ Bundle.prototype.addFile = function(name, options, cb) {
   });
 };
 
-Bundle.prototype.write = function(file, cb) {
-  this.encode(function(err, png, original) {
-    if (err) return cb(err);
-    fs.writeFile(file, png, function(err) {
-      if (err) return cb(err);
-      cb(null, png.length, original.length);
-    });
-  });
-};
-
-Bundle.prototype.encode = function(cb) {
+Bundle.prototype.write = function(filename, cb) {
   var channels = 3,
     buffer = this.toBuffer(),
     pixelCount = Math.ceil(buffer.length / channels),
     width = Math.ceil(Math.sqrt(pixelCount)),
     height = Math.ceil(pixelCount / width),
-    png = new PngEncoder(buffer, width, height, 'rgb');
+    png = new nodePng.PNG({width: width, height: height}),
+    i, srcIdx, dstIdx, file;
 
-  png.encode(function(data, err) {
-    cb(err, data, buffer);
+  for (i = 0; i < pixelCount; i++) {
+    srcIdx = i * 3;
+    dstIdx = i * 4;
+    png.data[dstIdx] = buffer[srcIdx];
+    png.data[dstIdx + 1] = buffer[srcIdx + 1];
+    png.data[dstIdx + 2] = buffer[srcIdx + 2];
+    png.data[dstIdx + 3] = 255;
+  }
+
+  file = fs.createWriteStream(filename);
+  file.on('close', function() {
+    cb(null, file.bytesWritten, png.data.length);
   });
+  png.pack().pipe(file);
 };
 
 // Convert a bundle to a buffer, containing header data and the contents of each file

